packages/shared/src/components/modals/referral/SearchReferralModal.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { ReactElement } from 'react';
+import classNames from 'classnames';
+import { Modal, ModalProps } from '../common/Modal';
+import { Checkbox } from '../../fields/Checkbox';
+import { Button } from '../../buttons/Button';
+import CopyIcon from '../../icons/Copy';
+import { useSettingsContext } from '../../../contexts/SettingsContext';
+import { CampaignCtaPlacement } from '../../../graphql/settings';
+import useMedia from '../../../hooks/useMedia';
+import { laptop } from '../../../styles/media';
+import { cloudinary } from '../../../lib/image';
+import { KeysRow } from './KeysRow';
+import { ReferralCampaignKey, useReferralCampaign } from '../../../hooks';
+import { link } from '../../../lib/links';
+import { useCopyLink } from '../../../hooks/useCopy';
+import { useAnalyticsContext } from '../../../contexts/AnalyticsContext';
+import { AnalyticsEvent, TargetType } from '../../../lib/analytics';
+import CloseButton from '../../CloseButton';
+
+function SearchReferralModal({
+  onRequestClose,
+  ...props
+}: ModalProps): ReactElement {
+  const isLaptop = useMedia([laptop.replace('@media ', '')], [true], false);
+  const { trackEvent } = useAnalyticsContext();
+  const { campaignCtaPlacement, onToggleHeaderPlacement } =
+    useSettingsContext();
+  const { availableCount, noKeysAvailable, url, referralToken } =
+    useReferralCampaign({
+      campaignKey: ReferralCampaignKey.Search,
+    });
+  const [isCopying, copyLink] = useCopyLink(() => url);
+  const handleCopy = () => {
+    trackEvent({
+      event_name: AnalyticsEvent.CopyKeyLink,
+      extra: JSON.stringify({ token: referralToken }),
+    });
+    copyLink();
+  };
+
+  const handleToggle = () => {
+    trackEvent({
+      event_name: AnalyticsEvent.HideFromHeader,
+      target_type: TargetType.HideInviteCheckbox,
+      target_id: !campaignCtaPlacement,
```

Comment:
Is this correct?
value states true/false, but is campaignCtaPlacement not a string in our case?

So the evaluation would not be correct I think